### DISCUSSION TITLE
Problems with recognition of leading '#', '##' to recognize sections

### DISCUSSION THREAD

**Discussion by friendly** [quarto-dev/quarto-cli#10252] [Q&A]
### Description

[Env: R 4.3.2, RStudio 2023.12.1+402, Quarto 1.3.450]

Quarto gets confused about what constitutes a chapter/section heading in a number of contexts, but the worst is when
a code chunk with a comment appears before the title of a chapter. 

To test what chunk options were in use in a chapter I inserted a chunk as follow before the chapter title.
What got used as the chapter title there was the comment **# what are figure defaults**

I know the simple fix: DONT DO THAT -- move the chunk after the title, but it was unnerving.
It seems like scanning for ch/section titles should ignore comments within chunks. Using `eval:false` had no effect

````
```{r include=FALSE}
source("R/common.R")
knitr::opts_chunk$set(fig.path = "figs/ch06/")
```

{{< include latex/latex-commands.qmd >}}

```{r}
#| eval: false
#| echo: false
# what are figure defaults?
knitr::opts_chunk$get(c("fig.width", "fig.height", "fig.align")) |> unlist()
```

# Plots for univariate response models {#sec-linear-models-plots}
````

I'm calling these "infelicities" not bugs, but it would be nice if these aspects were documented somewhere.

Another place this occurs is in fenced divs with an inline title:

```
::: {.callout-note}
# History corner
 ...
:::
```

Here, History corner appears as a section in the Outline view. I learned that I can avoid this using a title

```
::: {.callout-note title="History Corner"}
 ...
:::
```

**Comment by mcanouil** [comment]
As documented, title in call-out is a level two header not level one (level one is supposed to be document's title in markdown). https://quarto.org/docs/authoring/callouts.html

Regarding the automatic title detection, that might be undocumented for the knitr engine indeed. And possibly a bug if confirmed on latest Quarto version.

Also, I would suggest to upgrade to 1.5 to confirm the behaviour as you are running on 1.3 which is starting to get quite old.

**Reply by mcanouil** [reply]
Since your example is not reproducible I tried with a more reproducible version using latest development version of Quarto and did not see any issues:

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
```{r}
#| include: false
```

```{r}
#| eval: false
#| echo: false
# what are figure defaults?
1 + 1
```

# Plots for univariate response models

````

</td><td>

<img width="708" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/001355e7-846c-4853-8a02-d96cf64990eb">

</td></tr>
</table>

**Comment by cscheid** [comment]
I'll note, in passing, that `Quarto 1.3.450` is two stable versions old; the latest stable version is `1.5.54`, and I'd encourage you to update if you can.

**Comment by friendly** [comment]
Will do
